### PR TITLE
React testing

### PR DESCRIPTION
**Pull request by astroash**
Related to #448

#### Changes
- fix App layout post removal of prettier
- add component tests for homepage
- move error message outside of label (element cannot be grabbed for testing from inside)
- create renderWithRouter helper to allow testing with faux history object
- test how component responds to successful & unsuccessful statuscodes

### CODE REVIEW

File: client/src/cb-admin/pages/Login.js [hunk 1]
```diff
@@ -40,19 +39,20 @@ export default class Login extends React.Component {
 
     CbAdmin.login({ email: this.state.email, password: this.state.password })
       .then((res) => {
-        localStorage.setItem('token', res.data.result.token);
+        if (process.env.NODE_ENV !== 'test') localStorage.setItem('token', res.data.result.token);
```
Comment (eliascodes) [inline client/src/cb-admin/pages/Login.js]: Is this stripped out by babel at build time? `process` is a Node API and will cause an exception on the browser.

Comment (astroash) [reply client/src/cb-admin/pages/Login.js]: This was working fine locally, (ie no errors and you can actually console.log(process.env) in the browser), but I guess this is something to do with CRA's hot reloading. Any suggestions of a better approach? if not I could probs do:
```
if(process && process...)
```

**Comment by eliascodes** [comment]
Oh man, such a big hit to coverage 😓

**Comment by astroash** [comment]
> Oh man, such a big hit to coverage 😓

@eliascodes technically, we've doubled the codebase being monitored for coverage, and have _just about_ dropped by less than half...

I see this as a success 🎉

File: .travis.yml [hunk 2]
```diff
@@ -28,6 +28,8 @@ script:
   - npm run build.client
   - npm run cover
   - npm run cover.report.ci
+  - npm run test:client
+  - npm run cover:client
```
Comment (eliascodes) [inline .travis.yml]: Why run both test and cover?

File: client/src/cb-admin/pages/__tests__/Login.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import MockAdapter from 'axios-mock-adapter';
+import renderWithRouter from '../../../tests';
+import Login from '../Login';
+
+
+describe('Login Component', () => {
+  afterEach(cleanup);
+
+  test(':: incorrect user details returns 401 and displays error message', async () => {
+    const mock = new MockAdapter(axios);
```
Comment (eliascodes) [inline client/src/cb-admin/pages/__tests__/Login.test.js]: Do we need to create a new instance of the mock adapter for each test case? Can we just re-use it?

**Pull request merged**